### ISSUE TITLE
Focus indicator should be rounded for submenu items

### ISSUE THREAD

**Issue by evansjohnson** [palantir/blueprint#6890]
From https://github.com/palantir/blueprint/pull/6854#issuecomment-2200929002

no border-radius on submenu items:
![Screenshot 2024-07-02 at 5 16 11 PM](https://github.com/palantir/blueprint/assets/14102129/8748fc55-c356-4a78-88e6-8a2feb6bc015)

expected border-radius:
![Screenshot 2024-07-02 at 5 16 07 PM](https://github.com/palantir/blueprint/assets/14102129/f36520b0-11f4-4235-b7f7-d975acc80f8d)
